sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/MergeRowsExec.scala

```diff
@@ -233,6 +265,13 @@ case class MergeRowsExec(
         }
       }
 
+      // no instructions matched, drop the row
```

Comment:
Optional: I'd say having empty lines between each of these statements would better fit the existing style.

```
// no instructions matched, drop the row

if (targetPresent) {
  longMetric("numTargetRowsUnused") += 1
}

if (sourcePresent) {
  longMetric("numSourceRowsUnused") += 1
}

null
```